release/models/hashing/openconfig-hashing.yang

```diff
@@ -0,0 +1,176 @@
+module openconfig-hashing {
+    yang-version "1";
+
+    // namespace
+    namespace "http://openconfig.net/yang/hashing";
+
+    prefix "oc-hashing";
+
+    // import some basic types
+    import openconfig-extensions { prefix oc-ext; }
+    import openconfig-packet-match { prefix oc-pkt-match; }
+
+    // meta
+    organization "OpenConfig working group";
+
+    contact
+      "OpenConfig working group
+      netopenconfig@googlegroups.com";
+
+    description
+      "Model for managing hashing policies that would be referenced by the
+      interfaces model.";
+
+
+    oc-ext:openconfig-version "0.1.0";
+
+    revision "2023-08-08" {
+        description "Initial hashing model.";
+        reference "0.1.0";
+    }
+
+    // OpenConfig specific extensions for module metadata.
+    oc-ext:regexp-posix;
+    oc-ext:catalog-organization "openconfig";
+    oc-ext:origin "openconfig";
+
+    grouping hashing-inputs {
+        description
+          "Top level container for inputs to be used for hashing policies.";
+
+        container config {
+            description
+              "Configurable items at the hashing inputs level.";
+            container ipv4 {
+               uses oc-pkt-match:ipv4-protocol-fields-config;
```

Comment:
This grouping configures an ipv4 address value which I am not sure is what is needed?

Is the intent to select if ipv4 source and/or destination address fields should be used  for hash computation?  If yes, probably something like this will work:

hash-field-modes/hash-field-mode/ipv4 =  enum(none, src_address, dst_address, both, target_defined)
hash-field-modes/hash-field-mode/ipv6 =  enum(none, src_address, dst_address, both,  flow_label,  target_defined)
hash-field-modes/hash-field-mode/ipv6 =  enum(none, src_address, dst_address, both,  target_defined)

hash-field-modes/hash-field-mode/transport_ports =  enum(none, srcport, dstport, both,  target_defined)
hash-field-modes/hash-field-mode/transport_ports =  enum(none, srcport, dstport, both,  target_defined)